pkg/components/chyt.go

```diff
@@ -34,13 +34,14 @@ func NewChyt(
 	chyt *apiproxy.Chyt,
 	ytsaurus *ytv1.Ytsaurus) *Chyt {
 
-	l := labeller.Labeller{
-		ObjectMeta:     &chyt.GetResource().ObjectMeta,
-		APIProxy:       chyt.APIProxy(),
-		ComponentLabel: fmt.Sprintf("ytsaurus-chyt-%s", chyt.GetResource().Name),
-		ComponentName:  fmt.Sprintf("CHYT-%s", chyt.GetResource().Name),
-		Annotations:    ytsaurus.Spec.ExtraPodAnnotations,
-	}
+	l := labeller.NewLabeller(
+		&chyt.GetResource().ObjectMeta,
+		consts.CHYTType,
+		consts.YTComponentLabelCHYT,
+	).
+		WithComponentInstanceName(chyt.GetResource().Name).
+		WithAnnotations(ytsaurus.Spec.ExtraPodAnnotations).
+		Build()
```

Comment:
Half constructor, half builder. Looks really ugly.

Reply:
As far as I understood we've agreed to have two (or more) constructors to cover of possible cases of creation with fields required for that case..